### PR TITLE
feat(sql) SQL for bulk insert

### PR DESCRIPTION
**Pull request by mkaruza**
Support for multiple row inserts.

### CODE REVIEW

**Comment by bluestreak01** [comment]
thank you @mkaruza 👍 

However the implemented functionality requires way more tests that is offered in the PR

**Comment by mkaruza** [comment]
Added new test cases

File: core/src/main/java/io/questdb/griffin/SqlCompiler.java [hunk 1]
```diff
@@ -1796,84 +1797,92 @@ private CompiledQuery insert(ExecutionModel executionModel, SqlExecutionContext
...
-
                     } else {
-                        throw SqlException.invalidColumn(model.getColumnPosition(i), columnSet.get(i));
+                        final int columnCount = metadata.getColumnCount();
+                        final ObjList<ExpressionNode> values = model.getRowValues(t);
+                        final int valueCount = values.size();
+                        if (columnCount != valueCount) {
+                            throw SqlException.$(
+                                    model.getEndOfValuesPosition(),
+                                    "not enough values [expected=").put(columnCount).put(", actual=").put(values.size())
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlCompiler.java:1848]: this error message is factually incorrect when row group has more values than target columns

File: core/src/main/java/io/questdb/griffin/SqlCompiler.java [hunk 2]
```diff
@@ -1796,84 +1797,92 @@ private CompiledQuery insert(ExecutionModel executionModel, SqlExecutionContext
...
+                                    .put(", row=").put(t+1).put(']');
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlCompiler.java:1849]: inconsistent naming row/tuple

File: core/src/main/java/io/questdb/griffin/SqlCompiler.java [hunk 3]
```diff
@@ -1796,84 +1797,92 @@ private CompiledQuery insert(ExecutionModel executionModel, SqlExecutionContext
...
                         }
-
                     } else {
-                        throw SqlException.invalidColumn(model.getColumnPosition(i), columnSet.get(i));
+                        final int columnCount = metadata.getColumnCount();
+                        final ObjList<ExpressionNode> values = model.getRowValues(t);
+                        final int valueCount = values.size();
+                        if (columnCount != valueCount) {
+                            throw SqlException.$(
+                                    model.getEndOfValuesPosition(),
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlCompiler.java:1847]: position does not reflect site of error

File: core/src/main/java/io/questdb/griffin/SqlCompiler.java [hunk 4]
```diff
@@ -1796,84 +1797,92 @@ private CompiledQuery insert(ExecutionModel executionModel, SqlExecutionContext
...
+                                        i,
+                                        i,
+                                        function,
+                                        node.position,
+                                        executionContext.getBindVariableService()
+                                );
+
+                                if (writerTimestampIndex == i) {
+                                    timestampFunction = function;
+                            }
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlCompiler.java:1872]: minor: code formatting

File: core/src/main/java/io/questdb/griffin/SqlCompiler.java [hunk 5]
```diff
@@ -2022,9 +2031,13 @@ private ExecutionModel lightlyValidateInsertModel(InsertModel model) throws SqlE
             throw SqlException.$(tableName.position, "literal expected");
         }
 
+        int rowTupleCount = model.getRowTupleCount();
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlCompiler.java:2034]: inconsistent naming: rowValues/rowTuples/rowValueTuples

File: core/src/main/java/io/questdb/griffin/model/InsertModel.java [hunk 6]
```diff
@@ -158,16 +163,18 @@ public void toSink(CharSink sink) {
         if (queryModel != null) {
             queryModel.toSink(sink);
         } else {
-            sink.put("values (");
-
-            for (int i = 0, m = columnValues.size(); i < m; i++) {
-                if (i > 0) {
-                    sink.put(", ");
+            sink.put("values ");
+            for (int t = 0; t < rowValuesTuples.size(); t++) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/model/InsertModel.java:167]: loop syntax

File: core/src/main/java/io/questdb/griffin/SqlCompiler.java [hunk 7]
```diff
@@ -2022,9 +2031,13 @@ private ExecutionModel lightlyValidateInsertModel(InsertModel model) throws SqlE
...
         int columnSetSize = model.getColumnSet().size();
-        if (columnSetSize > 0 && columnSetSize != model.getColumnValues().size()) {
-            throw SqlException.$(model.getColumnPosition(0), "value count does not match column count");
+
+        for (int i = 0; i < rowTupleCount; i++) {
+            if (columnSetSize > 0 && columnSetSize != model.getRowValues(i).size()) {
+                throw SqlException.$(model.getColumnPosition(0), "row value count does not match column count [tuple=" + (i + 1) + "]");
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/SqlCompiler.java:2039]: error position is incorrect - should position on tuple with incorrect value count

**Review by bluestreak01 — CHANGES_REQUESTED** [review]
In addition to comments above, the following syntax problems exist:

```sql
create table x(a int, b int, c int);
```

```sql
-- expected error got 1 row + success
insert into x values ((1,2,3), (1,2), (3,4,5));
```

```sql
-- insert executes when syntax error is expected
insert into x values (1,3,5),(3,4,0),;
```

```sql
-- error text is factually incorrect, error position is incorrect
insert into x values (1,3,5,6),(3,4,0), (5,6,9);
```

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**